nrf52-hal-common/src/spim.rs

```diff
@@ -155,25 +156,58 @@ impl<T> Spim<T> where T: SpimExt {
             w.enable().enabled()
         );
 
-        // Set to SPI mode 0
-        spim.config.write(|w|
-            w
-                .order().msb_first()
-                .cpha().leading()
-                .cpol().active_high()
-        );
+        // Configure mode
+        spim.config.write(|w| {
+            // Can't match on `mode` as embedded_hal::spi::Mode only derives PartialEq, not Eq
+            if mode == MODE_0 {
+                w.order().msb_first().cpol().active_high().cpha().leading()
+            } else if mode == MODE_1 {
+                w.order().msb_first().cpol().active_high().cpha().trailing()
+            } else if mode == MODE_2 {
+                w.order().msb_first().cpol().active_low().cpha().leading()
+            } else {
+                w.order().msb_first().cpol().active_low().cpha().trailing()
```

Comment:
Maybe panic or return Err here if not mode 0..3?

Reply:
There are only 4 modes possible (parity and phase are booleans), so if it is not mode 0 through 2, it must be 3, and there are no other possibilities. I agree this is ugly and unclear, adding the `Eq` derive would mean being able to use a match which would be much nicer.

Reply:
I would prefer a match here ;) It's the rusty way to do this :)

Reply:
We can't match on the value since it doesn't implement `Eq`. The next embedded-hal release will fix that, but I'd prefer not to wait for that.

```
error: to use a constant of type `X` in a pattern, `X` must be annotated with `#[derive(PartialEq, Eq)]`
```

Reply:
Oh, I am very sorry, it's even written down below ... I didn't read on first ... I somehow thought it's an enum.